vulnerabilities/importers/pysec.py

```diff
@@ -0,0 +1,127 @@
+# derivative work, you must accompany this data with the following acknowledgment:
+#
+#  Generated with VulnerableCode and provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+#  OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+#  VulnerableCode should be considered or used as legal advice. Consult an Attorney
+#  for any legal advice.
+#  VulnerableCode is a free software code scanning tool from nexB Inc. and others.
+#  Visit https://github.com/nexB/vulnerablecode/ for support and download.
+import json
+import logging
+from datetime import datetime
+from datetime import timezone
+from io import BytesIO
+from typing import Iterable
+from zipfile import ZipFile
+
+import requests
+from packageurl import PackageURL
+from univers.version_range import GitHubVersionRange
+from univers.version_range import PypiVersionRange
+from univers.versions import SemverVersion
+from univers.versions import Version
+
+from vulnerabilities.importer import AdvisoryData
+from vulnerabilities.importer import AffectedPackage
+from vulnerabilities.importer import Importer
+from vulnerabilities.importer import Reference
+from vulnerabilities.importer import VulnerabilitySeverity
+from vulnerabilities.severity_systems import SCORING_SYSTEMS
+
+logger = logging.getLogger(__name__)
+
+
+class PyPlImporter(Importer):
+    spdx_license_expression = "Apache-2.0"
+
+    def advisory_data(self) -> Iterable[AdvisoryData]:
+        url = "https://osv-vulnerabilities.storage.googleapis.com/PyPI/all.zip"
+        response = requests.get(url).content
+        try:
+            with ZipFile(BytesIO(response)) as zip_file:
+                for file_name in zip_file.namelist():
+                    with zip_file.open(file_name) as f:
+                        vul_info = json.loads(f.read())
+                        yield self.parse_advisory_data(vul_info)
+        except requests.exceptions.RequestException:
+            logger.error(f"Failed to fetch osv-vulnerabilities PyPI: HTTP ")
+
+    @staticmethod
+    def parse_advisory_data(raw_data: dict) -> AdvisoryData:
+        summary = raw_data["summary"] if "summary" in raw_data else raw_data["details"][slice(50)]
+        aliases = raw_data["aliases"] if "aliases" in raw_data else raw_data["id"]
+
+        date_published = datetime.strptime(
+            raw_data["published"][0:19], "%Y-%m-%dT%H:%M:%S"
+        ).replace(tzinfo=timezone.utc)
+
+        severity = []
+        if "ecosystem_specific" in raw_data and "severity" in raw_data["ecosystem_specific"]:
+            severity = VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["generic_textual"],
+                value=raw_data["ecosystem_specific"]["severity"],
+            )
+
+        references = []
+        for ref in raw_data["references"]:
+            references.append(Reference(url=ref["url"], severities=severity))
+
+        affected_package = []
+        # fixed_version = TODO
```

Comment:
I don't know what is the best type for fixed version

Reply:
@ziadhany The current design is to use an AffectedPackage: with the fixed version set.
See https://github.com/nexB/vulnerablecode/blob/4390b6aca872cbde378516199f143fa11e64bc5a/vulnerabilities/importer.py#L116